### PR TITLE
Numerous fixes

### PR DESCRIPTION
**Pull request by AlexDenisov**

### CODE REVIEW

File: lib/GoogleTest/GoogleTestMutationOperatorFilter.cpp [hunk 1]
```diff
@@ -28,6 +28,10 @@ bool GoogleTestMutationOperatorFilter::shouldSkipTesteeFunction(llvm::Function *
     return true;
   }
 
+  if (testee->getName().find(StringRef("clang_call_terminate")) != StringRef::npos) {
```
Comment (stanislaw) [inline lib/GoogleTest/GoogleTestMutationOperatorFilter.cpp]: +1, saw this one many times.

File: lib/GoogleTest/GoogleTestFinder.cpp [hunk 2]
```diff
@@ -109,12 +111,14 @@ std::vector<std::unique_ptr<Test>> GoogleTestFinder::findTests(Context &Ctx) {
...
       /// compare the beginning of the typename
 
-      if (!STy->getName().startswith(StringRef("class.testing::TestInfo"))) {
+      if (!structType->getName().startswith(testInfoTypeName)) {
         continue;
       }
 
-      /// At this point the Global has only one usage
-      /// The user of the Global is part of initialization function
+      /// Normally the globalValue has only one usage, ut LLVM could add
```
Comment (stanislaw) [inline lib/GoogleTest/GoogleTestFinder.cpp]: typo :trollface:

File: lib/GoogleTest/GoogleTestFinder.cpp [hunk 3]
```diff
@@ -109,12 +111,14 @@ std::vector<std::unique_ptr<Test>> GoogleTestFinder::findTests(Context &Ctx) {
       /// Hence we cannot just compare string, and rather should
       /// compare the beginning of the typename
 
-      if (!STy->getName().startswith(StringRef("class.testing::TestInfo"))) {
```
Comment (stanislaw) [inline lib/GoogleTest/GoogleTestFinder.cpp]: 👍

File: lib/GoogleTest/GoogleTestFinder.cpp [hunk 4]
```diff
@@ -228,19 +238,6 @@ std::vector<std::unique_ptr<Test>> GoogleTestFinder::findTests(Context &Ctx) {
   return tests;
 }
 
-std::string demangle(const char* name) {
```
Comment (stanislaw) [inline lib/GoogleTest/GoogleTestFinder.cpp]: 👍 but we might need it for a display name later on in some other class (to display logs for example?).

Comment (AlexDenisov) [inline lib/GoogleTest/GoogleTestFinder.cpp]: Then we would add it :)

File: lib/GoogleTest/GoogleTestFinder.cpp (hunk 2 above)
Comment (AlexDenisov) [inline lib/GoogleTest/GoogleTestFinder.cpp]: @stanislaw open a ticket ;)

Comment (stanislaw) [inline lib/GoogleTest/GoogleTestFinder.cpp]: Will do thanks for the hint 👍

**Pull request merged**